testing/regress/ecl/parquetTypes.ecl

```diff
@@ -0,0 +1,617 @@
+    {172, 'ado', -1.41421356237309D}
+], real8Record);
+
+ParquetIO.Write(real8DatasetOut, dropzoneDirectory + '/Real8Test.parquet', TRUE);
+
+real8DatasetIn := ParquetIO.Read(real8Record, dropzoneDirectory + '/Real8Test.parquet');
+
+{UNSIGNED testid, STRING3 testname, BOOLEAN isEqual} real8Join (real8DatasetOut a, real8DatasetIn b) := TRANSFORM
+    SELF.testid := a.testid;
+    SELF.testname := a.testname;
+    SELF.isEqual := a.value = b.value;
+END;
+
+real8CompareResult := JOIN(real8DatasetOut, real8DatasetIn, LEFT.testid = RIGHT.testid, real8Join(LEFT, RIGHT), ALL);
+
+real8Result := IF(COUNT(real8CompareResult(isEqual = FALSE)) = 0, 'Pass', 'Fail: REAL8 data mismatch');
+
+// ======================== SET OF STRING ========================
+
+setOfStringRecord := {UNSIGNED testid, STRING3 testname, SET OF STRING value};
+
+setOfStringDatasetOut := DATASET([
+    {180, 'adp', ['Set', 'Of', 'String', 'Test']},
+    {181, 'adq', ['ECL', 'Data', 'Types']},
+    {184, 'adt', ['A', 'B', 'C', 'D', 'E']}
+], setOfStringRecord);
+
+ParquetIO.Write(setOfStringDatasetOut, dropzoneDirectory + '/SetOfStringTest.parquet', TRUE);
+
+setOfStringDatasetIn := ParquetIO.Read(setOfStringRecord, dropzoneDirectory + '/SetOfStringTest.parquet');
+
+{UNSIGNED testid, STRING3 testname, BOOLEAN isEqual} setOfStringJoin (setOfStringDatasetOut a, setOfStringDatasetIn b) := TRANSFORM
+    SELF.testid := a.testid;
+    SELF.testname := a.testname;
+    SELF.isEqual := a.value = b.value; // Compare SET OF STRING values
+END;
+
+setOfStringCompareResult := JOIN(setOfStringDatasetOut, setOfStringDatasetIn, LEFT.testid = RIGHT.testid, setOfStringJoin(LEFT, RIGHT), ALL);
+
+setOfStringResult := IF(COUNT(setOfStringCompareResult(isEqual = FALSE)) = 0, 'Pass', 'Fail: SET OF STRING data mismatch');
+
+// ======================== SET OF UNICODE ========================
+
+setOfUnicodeRecord := {UNSIGNED testid, STRING3 testname, SET OF UNICODE value};
+
+setOfUnicodeDatasetOut := DATASET([
+    {192, 'adw', [U'Á', U'É', U'Í', U'Ó', U'Ú']},
+    {193, 'adx', [U'α', U'β', U'γ', U'δ', U'ε']},
+    {194, 'ady', [U'☀', U'☁', U'☂', U'☃', U'☄']}
+], setOfUnicodeRecord);
+
+ParquetIO.Write(setOfUnicodeDatasetOut, dropzoneDirectory + '/SetOfUnicodeTest.parquet', TRUE);
+
+setOfUnicodeDatasetIn := ParquetIO.Read(setOfUnicodeRecord, dropzoneDirectory + '/SetOfUnicodeTest.parquet');
+
+{UNSIGNED testid, STRING3 testname, BOOLEAN isEqual} setOfUnicodeJoin (setOfUnicodeDatasetOut a, setOfUnicodeDatasetIn b) := TRANSFORM
+    SELF.testid := a.testid;
+    SELF.testname := a.testname;
+    SELF.isEqual := a.value = b.value; 
+END;
+
+setOfUnicodeCompareResult := JOIN(setOfUnicodeDatasetOut, setOfUnicodeDatasetIn, LEFT.testid = RIGHT.testid, setOfUnicodeJoin(LEFT, RIGHT), ALL);
+
+setOfUnicodeResult := IF(COUNT(setOfUnicodeCompareResult(isEqual = FALSE)) = 0, 'Pass', 'Fail: SET OF UNICODE data mismatch');
+
+// ======================== INTEGER8 ========================
```

Comment:
Same comment as above. Add repeating types to the first section of that type as an additional field. See the DATA section as an example on line 173.

Reply:
To avoid making too many comments I explicitly mark each occurrence, but ensure that each section cannot be moved to a preexisting one.